gitlab/tests/test_integration.py

```diff
@@ -7,32 +7,54 @@
 
 from datadog_checks.gitlab import GitlabCheck
 
-from .common import AUTH_CONFIG, BAD_CONFIG, CUSTOM_TAGS, HOST
+from .common import CUSTOM_TAGS, HOST, METRICS_TO_TEST, assert_check
 
 pytestmark = [pytest.mark.usefixtures("dd_environment"), pytest.mark.integration]
 
 
-def test_connection_failure(aggregator):
+def test_check(dd_run_check, aggregator, gitlab_check, config):
     """
     Make sure we're failing when the URL isn't right
     """
+    check = gitlab_check(config)
+    dd_run_check(check)
 
-    gitlab = GitlabCheck('gitlab', BAD_CONFIG['init_config'], instances=BAD_CONFIG['instances'])
+    assert_check(aggregator, METRICS_TO_TEST)
+
+
+def test_connection_failure(aggregator, gitlab_check, bad_config):
+    """
+    Make sure we're failing when the URL isn't right
+    """
+
+    check = gitlab_check(bad_config)
 
     with pytest.raises(ConnectionError):
-        gitlab.check(BAD_CONFIG['instances'][0])
+        check.check(None)
 
-    # We should get only one failed service check, the first
     aggregator.assert_service_check(
-        'gitlab.{}'.format(GitlabCheck.ALLOWED_SERVICE_CHECKS[0]),
+        'gitlab.prometheus_endpoint_up',
+        status=GitlabCheck.CRITICAL,
+        count=1,
+    )
```

Comment:
To reviewer: I'll add the tag in a separate PR because this is a bug https://github.com/DataDog/integrations-core/pull/14262